### PR TITLE
Transpose throws an annoying wrench in the mix

### PR DESCRIPTION
**Pull request by drisspg**
# Summary

When you call torch.nn.F.linear() you will call transpose on the weight.

One solution for doing this is to lazily compute the transpose, which is mark that the matrix needs to be transposed when it encounters a realizing op. In this case that would be addmm and mm. Did not handle correctly double even transposes cause I think that's unlikely

### CODE REVIEW

**Review by rohan-varma — COMMENTED** [review]
Curious, where is this extra info being stored?

**Comment by drisspg** [comment]
https://github.com/pytorch-labs/float8_experimental/blob/main/float8_experimental/float8_ops.py#L44

right now we have this "emulate" field where whenver you write an op you have to make sure that emulate gets set correctly on the newly created Float8Tensor.

I just did something similar for NF4 and that works, but I feels kinda code smelly to me. Its essentially a lazy transpose but since we only need to support these 3 ops it should work.

**Comment by drisspg** [comment]
@rohan-varma Okay I added this and it works, need to add compile tests and the do the memory profiling things

File: transformer_nuggets/quant/nf4_tensor.py [hunk 1]
```diff
@@ -33,6 +33,39 @@ class SubclassTensorArgs:
     requires_grad: bool
 
 
+@implements([aten.t.default])
+def nf4_transpose(aten_op, args, kwargs=None):
+    """Transpose for NF4 Tensors"""
+    # We are arused the input is 2 dimensional otherwise we would
```
Comment (drisspg) [inline transformer_nuggets/quant/nf4_tensor.py:39]: insured

**Pull request closed without merging**